### PR TITLE
(PDK-1250) Adds MacOS 10.14 packaging

### PR DESCRIPTION
**Pull request by bmjen**

### CODE REVIEW

File: configs/platforms/osx-10.14-x86_64.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    plat.provision_with 'mkdir -p /etc/homebrew'
+    plat.provision_with 'cd /etc/homebrew'
+    plat.provision_with 'su test -c \'echo | /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"\''
+    plat.provision_with 'sudo chown -R test:admin /Users/test/Library/'    
+  
+    packages = %w[cmake pkg-config yaml-cpp]
+  
+    plat.provision_with "su test -c '/usr/local/bin/brew install #{packages.join(' ')}'"
+  
+    plat.vmpooler_template 'osx-1012-x86_64'
```
Comment (scotje) [inline configs/platforms/osx-10.14-x86_64.rb]: Can we use the `osx-1014-x86_64` vmpooler template?

Comment (bmjen) [reply configs/platforms/osx-10.14-x86_64.rb]: doh. Missed that. Let me retest.

Comment (scotje) [reply configs/platforms/osx-10.14-x86_64.rb]: Yeah, I think with 10.13 there was some reason it was still using the 10.12 template, so I'm curious if that is still necessary.

Comment (bmjen) [reply configs/platforms/osx-10.14-x86_64.rb]: I updated it to use 1014 image, and tested the packaging locally and it worked fine.

**Pull request merged**